src/researchhub_document/views/researchhub_post_views.py

```diff
@@ -103,41 +101,39 @@ def upsert_researchhub_posts(self, request):
         except (KeyError, TypeError) as exception:
             return Response(exception, status=400)
 
+    def _check_authors_in_org(self, authors, organization):
+        for author_id in authors:
+            author = User.objects.get(id=author_id)
+            if not organization.org_has_user(author):
+                return False
+        return True
```

Comment:
I ran into an issue while testing. This line in particular
```
User.objects.get(id=author_id)
```

Throws an error because an `author_id` is passed instead of a user_id.
I think this line should be updated to  
```
Author.objects.get(id=author_id)
````

There is a 1:1 relationship between author and user. It seems like author ids are passed in the request and not user ids. This may have not errored out in your local env because locally, it is very likely that the author ids and user ids are identical since they are created sequentially

Reply:
Thanks for pointing this out. Correct, I haven't noticed this in my local environment due to user IDs and author IDs aligning.